soroban-simulation/src/test/simulation.rs

```diff
@@ -478,10 +478,10 @@ fn test_simulate_invoke_contract_with_auth() {
     assert!(!res.diagnostic_events.is_empty());
 
     let (expected_instructions, expected_read_bytes, expected_resource_fee, expected_mem_bytes) =
-        if ledger_info.protocol_version == CURRENT_PROTOCOL_VERSION {
-            (38210172, 7492, 76472, 19104958)
-        } else {
-            (41268119, 7540, 79532, 20633919)
+        match ledger_info.protocol_version {
+            20 => (38210172, 7492, 76472, 19104958),
```

Comment:
v20 is no longer interesting for the simulation. We should bump `CURRENT_PROTOCOL_VERSION` to 21 instead and update the if statement respectively (that's why that's not a match in the first place).

Reply:
Even when doing replay? I think we want stability.

Reply:
Simulation doesn't replay anything, it's only for the interactions that haven't happened yet, so only the 'current' ledger state is relevant.

Reply:
ok, fixed

Reply:
actually I had to revert this back to the version you reviewed here -- we run the tests under all protocols as part of CI, even if simulation tests don't _need_ it, carving them out of the CI is more trouble than it's worth.